container-host-files/etc/scf/config/scripts/forward_logfiles.sh

```diff
@@ -1,175 +1,171 @@
+                :app-name, startswith, "vcap." @${SCF_LOG_PREFIX}${SCF_LOG_HOST}:${SCF_LOG_PORT};RFC5424Format
+                :app-name, startswith, "vcap." ~                # Stop writing SCF message logs to /var/log
 	EOF
-                echo "Rsyslog forwarder: Could not create $MAIN_CONFIG in $RSYSLOG_CONF_DIR" >> $PB_OUT
+                echo "Rsyslog forwarder: Could not create ${MAIN_CONFIG} in ${RSYSLOG_CONF_DIR}" >> "${PB_OUT}"
                 exit 0
         fi
 
-        if [[ ! -f "$RSYSLOG_CONF_DIR/$MAIN_CONFIG" ]]; then
-                echo "Rsyslog forwarder: File $MAIN_CONFIG not found in $RSYSLOG_CONF_DIR" >> $PB_OUT
+        if [[ ! -f "${RSYSLOG_CONF_DIR}/${MAIN_CONFIG}" ]]; then
+                echo "Rsyslog forwarder: File ${MAIN_CONFIG} not found in ${RSYSLOG_CONF_DIR}" >> "${PB_OUT}"
                 exit 0
         fi
 }
 
 # check if more logs to be monitored by rsyslog have come into existence since the last run
 function searchTargetDir {
-        filesAdded=1
-        for file in $1/*
-        do
-                if [ -d $file ]; then
-                        ignored=false
-                        for ignore in $IGNORE_DIR
-                        do
-                                if [[ $file == $RSYSLOG_FORWARDER_WATCH_DIR/$ignore ]]; then
-                                        echo "Ignoring $file directory"
-                                        ignored=true
-                                fi
-                        done;
-                        if [ $ignored == false ]; then
-                                if searchTargetDir $file; then
-                                        filesAdded=0
-                                fi
-                        fi
+        local dir file filesAdded=1
+        local args=( "$1" )
+        for dir in ${IGNORE_DIR} ; do
+                args=( "${args[@]}" '(' -name "${dir}" -a -prune ')' -o )
+        done
+        args=( "${args[@]}" '(' -name '*.log' -a -type f -a -print0 ')' )
+        while IFS= read -r -d '' file ; do
+                TARGET_BASENAME="$(basename "$1" .log)"
+                TARGET_NAME="${RSYSLOG_CONF_DIR}/${RSYSLOG_CONF_PREFIX}-${TARGET_BASENAME}.conf"
+                if checkConfigExists ; then
+                        echo "${TARGET_NAME} exists"
                 else
-                        if [ "${file: -4}" == ".log" ]; then
-                                targetName $file
-                                if checkConfigExists $file; then
-                                        echo $TARGET_NAME exists
-                                else
-                                        echo "Creating $TARGET_NAME"
-                                        createTargetConf $file
-                                        filesAdded=0
-                                fi
-                        fi
+                        echo "Creating ${TARGET_NAME}"
+                        createTargetConf "${file}"
+                        filesAdded=0
```

Comment:
Is `filesAdded` like `exit 0/1` where `0 = ok/true`, and `1 = fail/false`, i.e. reverse sense ?

Reply:
Yes, it is. It's insane, but it was insane before my changes.

Reply:
Yep.